workflows/cloudify_system_workflows/snapshots/copy_icons.py

```diff
@@ -0,0 +1,55 @@
+import os
+import sys
+import shutil
+
+from manager_rest.flask_utils import setup_flask_app
+from manager_rest import config
+from manager_rest.constants import (FILE_SERVER_BLUEPRINTS_FOLDER,
+                                    BLUEPRINT_ICON_FILENAME)
+
+STAGE_ICONS_PATH = '/opt/cloudify-stage/dist/userData/blueprint-icons'
+
+
+def load_icons():
+    if not os.path.exists(STAGE_ICONS_PATH):
+        sys.stderr.write('No stage icons found, aborting icon copy.\n')
+        return
+
+    with setup_flask_app().app_context():
+        config.instance.load_configuration()
+        fs_blueprints_path = os.path.join(config.instance.file_server_root,
+                                          FILE_SERVER_BLUEPRINTS_FOLDER)
+    existing_blueprints = {}
+    for tenant in os.listdir(fs_blueprints_path):
+        tenant_path = os.path.join(fs_blueprints_path, tenant)
+        for blueprint in os.listdir(tenant_path):
```

Comment:
maybe an `os.isdir` somewhere here too?

Reply:
Hmm... I'm not sure that's required?

This'll only work on a node with stage on it where blueprint icon dump has occurred, which means we know it's a manager node (because at the moment they must be colocated).

Therefore, we know that the blueprints path must exist (I think it's created during bootstrap or thereabouts?).
If that's the case then because we create this path by listing the tenant directories in that dir, this dir must exist.

Any faults in that logic?